src/Auth_Command.php

```diff
@@ -0,0 +1,330 @@
+	 *
+	 * [<append>]
+	 * : Append ips in whitelisting of a site or globally.
+	 *
+	 * [<list>]
+	 * : List whitelisted ip's of a site or of global scope.
+	 *
+	 * [<remove>]
+	 * : Remove whitelisted ip's of a site or of global scope.
+	 *
+	 * [<site-name>]
+	 * : Name of website to be secured / `global` for global scope.
+	 *
+	 * [--ip=<ip>]
+	 * : Comma seperated ips.
+	 */
+	public function whitelist( $args, $assoc_args ) {
+
+		$commands = [ 'create', 'append', 'list', 'remove' ];
+		if ( ! ( isset( $args[0] ) && in_array( $args[0], $commands ) ) ) {
+			$help = PHP_EOL;
+			foreach ( $commands as $command ) {
+				$help .= "ee auth whitelist $command [<site-name>/global] [--ip=<ip>]" . PHP_EOL;
+			}
+			EE::error( 'Please use valid command syntax. You can use:' . $help );
+
+		}
+
+		$command = array_shift( $args );
+		$global  = $this->populate_info( $args, __FUNCTION__ . ' ' . $command );
+
+		$ip = EE\Utils\get_flag_value( $assoc_args, 'ip' );
+
+		$file         = EE_CONF_ROOT . '/nginx/vhost.d/';
+		$file         .= $global ? 'default_acl' : $this->db->site_url . '_acl';
+		$user_ips     = array_filter( explode( ',', $ip ), 'strlen' );
+		$existing_ips = $this->get_ips_from_file( $global );
+
+		call_user_func_array( [ $this, "whitelist_$command" ], [ $file, $user_ips, $existing_ips ] );
+
+		$this->reload();
+	}
+
+	/**
+	 * Function to create whitelist file.
+	 *
+	 * @param string $file        The whitelisting file.
+	 * @param array $user_ips     ip's provided by the user.
+	 * @param array $existing_ips Existing ip's in the given file.
+	 */
+	private function whitelist_create( $file, $user_ips, $existing_ips ) {
+
+		$file_content = '';
+		foreach ( $user_ips as $ip ) {
+			$file_content .= "allow $ip;" . PHP_EOL;
+		}
+		$this->fs->dumpFile( $file, $file_content );
+		EE::log( sprintf( 'Created whitelist for `%s` scope with %s IP\'s.', $this->db->site_url, implode( ',', $user_ips ) ) );
+	}
+
+	/**
+	 * Function to append to whitelist file.
+	 *
+	 * @param string $file        The whitelisting file.
+	 * @param array $user_ips     ip's provided by the user.
+	 * @param array $existing_ips Existing ip's in the given file.
+	 */
+	private function whitelist_append( $file, $user_ips, $existing_ips ) {
+
+		$all_ips      = array_unique( array_merge( $user_ips, $existing_ips ) );
+		$file_content = '';
+		foreach ( $all_ips as $individual_ip ) {
+			$file_content .= "allow $individual_ip;" . PHP_EOL;
+		}
+		$this->fs->dumpFile( $file, $file_content );
+		EE::log( sprintf( 'Appended %s IP\'s to whitelist of `%s` scope', implode( ',', $user_ips ), $this->db->site_url ) );
+	}
+
+	/**
+	 * Function to list whitelisted ips.
+	 *
+	 * @param string $file        The whitelisting file.
+	 * @param array $user_ips     ip's provided by the user.
+	 * @param array $existing_ips Existing ip's in the given file.
+	 */
+	private function whitelist_list( $file, $user_ips, $existing_ips ) {
+
+		if ( ! empty( $existing_ips ) ) {
```

Comment:
Just a suggestion: you can make this code simple...

if (  empty( $existing_ips ) ) { 
EE::error( sprintf( 'No Whitelisted IP\'s found for `%s` scope', $this->db->site_url ) );
}

EE::log( sprintf( 'Whitelisted IP\'s for `%s` scope', $this->db->site_url ) );
			foreach ( $existing_ips as $ips ) {
				EE::log( $ips );
			}

Reply:
Updated accordingly :+1: